src/Views/SearchView.vala

```diff
@@ -91,12 +100,22 @@ public class AppCenter.SearchView : Adw.NavigationPage {
             search_entry.grab_focus ();
         });
 
-        list_box.row_activated.connect ((row) => {
-            if (row is Widgets.PackageRow) {
-                show_app (((Widgets.PackageRow) row).get_package ());
-            }
+        factory.setup.connect ((obj) => {
+            var list_item = (Gtk.ListItem) obj;
+            list_item.child = new SearchListItem ();
+        });
+
+        factory.bind.connect ((obj) => {
+            var list_item = (Gtk.ListItem) obj;
+            ((SearchListItem) list_item.child).package = (AppCenterCore.Package) list_item.item;
+        });
+
+        list_view.activate.connect ((index) => {
+            show_app ((AppCenterCore.Package) list_store.get_item (index));
```

Comment:
Probably better to use the selection model here. Currently not a problem but when we use filtermodels and sorter models etc the original store doesn't match with the index anymore :)

Reply:
Good call!